src/main/java/com/rtb/projectmanagementtool/project/ProjectData.java

```diff
@@ -0,0 +1,208 @@
+/**
+ * ProjectData.java - this file implements the Project class, which contains details about a
+ * particular project, including the members and tasks within it.
+ */
+package com.rtb.projectmanagementtool.project;
+
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Entity;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+
+enum User {
+  CREATOR, // has admin capabilities and more, like deleting the project and add & remove admins
+  ADMIN, // can add users to the project/perform other administrative tasks
+  REGULAR;
+}
+
+public class ProjectData {
+  private final String PROPERTY_NAME = "name";
+  private final String PROPERTY_DESCRIPTION = "description";
+  private final String PROPERTY_USER_IDS = "userIds";
+  private final String PROPERTY_TASK_IDS = "taskIds";
+
+  private long id;
+  private String name;
+  private String description;
+  private HashMap<Long, User> userIds;
+  private HashSet<Long> taskIds;
+
+  /**
+   * Class constructor with the minimum requirements for creating a project.
+   *
+   * @param name the name of the project
+   * @param description the description of the project
+   * @param creatorId the id of the user who created the project
+   */
+  public ProjectData(String name, String description, long creatorId) {
+    this.name = name;
+    this.description = description;
+    this.taskIds = new HashSet<Long>();
+
+    this.userIds = new HashMap<Long, User>();
+    this.userIds.put(creatorId, User.CREATOR);
+  }
+
+  /**
+   * Class constructor to create Project from existing datastore entity
+   *
+   * @param entity the datastore entity to parse object from
+   */
+  public ProjectData(Entity entity) {
+    this.id = (Long) entity.getKey().getId();
+    this.name = (String) entity.getProperty(PROPERTY_NAME);
+    this.description = (String) entity.getProperty(PROPERTY_DESCRIPTION);
+
+    Object userIdProperty;
+    if ((userIdProperty = entity.getProperty(PROPERTY_TASK_IDS)) != null) {
+      this.taskIds = new HashSet<Long>((ArrayList<Long>) userIdProperty);
+    } else {
+      this.taskIds = new HashSet<Long>();
+    }
+
+    this.userIds = new HashMap<Long, User>();
+    // can store map in datastore entities using EmbeddedEntity
+    EmbeddedEntity ee = (EmbeddedEntity) entity.getProperty(PROPERTY_USER_IDS);
```

Comment:
Crafty usage of EmbeddedEntity to store maps.  Could be worth moving this into its own MapToEmbeddedEntity() method in a DataUtils.java file so everyone else can reuse.

Just keep in mind though that these are not searchable in a query.  Meaning you won't be able to do "get me projects owned by user X"

Reply:
I think I implemented the query functionality (unless I misunderstand what you mean by it) in getProjects(). It takes a Collection of userIds to fetch projects that only them are involved in. I could probably tweak the Collection type and my entityContainsUser() method to allow it to only grab projects owned by a user.